util/chunk/column.go

```diff
@@ -246,6 +240,89 @@ const (
 	sizeMyDecimal = int(unsafe.Sizeof(types.MyDecimal{}))
 )
 
+// preAlloc allocates space for a fixed-length-type slice and resets all slots to null.
+func (c *Column) preAlloc(length, typeSize int) {
+	nData := length * typeSize
+	if len(c.data) >= nData {
+		c.data = c.data[:nData]
+	} else {
+		c.data = make([]byte, nData)
+	}
+
+	nBitmap := (length + 7) >> 3
+	if len(c.nullBitmap) >= nBitmap {
+		c.nullBitmap = c.nullBitmap[:nBitmap]
+		for i := range c.nullBitmap {
+			// resets all slots to null.
+			c.nullBitmap[i] = 255
+		}
+	} else {
+		c.nullBitmap = make([]byte, nBitmap)
+	}
+
+	c.length = length
+}
+
+// SetNull sets the rowIdx to null.
+func (c *Column) SetNull(rowIdx int) {
+	if !c.IsNull(rowIdx) {
+		c.nullBitmap[rowIdx>>3] &= ^(1 << uint(rowIdx&7))
+	}
+}
+
+var (
+	bitCntMap [256]int
```

Comment:
There is `bits.OnesCount8` in the standard library.

Reply:
fixed.